### DISCUSSION TITLE
`render_template_as_native_obj` breaks posting data with `SimpleHttpOperator`

### DISCUSSION THREAD

**Discussion by sd274** [apache/airflow#24326] [General]
### Apache Airflow version

2.3.0

### What happened

I'm trying to create a dag that uses the `SimpleHttpOperator` to post data to various endpoints and pass the responses to other endpoints. As I am passing the responses to other endpoints it is convenient to have `render_template_as_native_obj=True`. However, when I have this set to True it seems to cause issues with the data that is posted. Example below (works when `render_template_as_native_obj=False`) the http_conn_id is set up with `https://jsonplaceholder.typicode.com/`):

```
from datetime import datetime
import yaml

from airflow import DAG
from airflow.operators.bash import BashOperator
from airflow.providers.http.operators.http import SimpleHttpOperator
from airflow.decorators import dag, task

dag_id = f'http_test'
@dag(dag_id=dag_id, start_date=datetime(2022, 6,7), render_template_as_native_obj=True)
def http_test():

    first_task = BashOperator(
        task_id='first_task',
        bash_command='echo helllo',
    )

    get_some_data_task = SimpleHttpOperator(
        task_id='get_some_data',
        method='POST',
        endpoint='/posts',
        http_conn_id='jsonplaceholder',
        headers={"Content-Type": "application/json"},
        data=json.dumps({
            'title': 'foo',
            'body': 'bar',
            'userId': 1,
        }),
    )

    first_task >> get_some_data_task

dag = http_test()
```

### What you think should happen instead

Having `render_template_as_native_obj` should only effect jinja templates, not other parts of the code.

### How to reproduce

Dag defined above works with `render_template_as_native_obj` as False, but fails with it as True.

### Operating System

mac-os

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon @ file:///pip-packages/apache_airflow_providers_amazon-3.4.0-py3-none-any.whl
apache-airflow-providers-celery==2.1.0
apache-airflow-providers-cncf-kubernetes==2.1.0
apache-airflow-providers-docker==2.3.0
apache-airflow-providers-elasticsearch==2.1.0
apache-airflow-providers-ftp @ file:///pip-packages/apache_airflow_providers_ftp-2.1.2-py3-none-any.whl
apache-airflow-providers-google==6.1.0
apache-airflow-providers-grpc==2.0.1
apache-airflow-providers-hashicorp==2.1.1
apache-airflow-providers-http @ file:///pip-packages/apache_airflow_providers_http-2.1.2-py3-none-any.whl
apache-airflow-providers-imap @ file:///pip-packages/apache_airflow_providers_imap-2.2.3-py3-none-any.whl
apache-airflow-providers-microsoft-azure==3.3.0
apache-airflow-providers-mysql==2.1.1
apache-airflow-providers-odbc==2.0.1
apache-airflow-providers-postgres==2.3.0
apache-airflow-providers-redis==2.0.1
apache-airflow-providers-sendgrid==2.0.1
apache-airflow-providers-sftp==2.2.0
apache-airflow-providers-slack==4.1.0
apache-airflow-providers-snowflake @ file:///pip-packages/apache_airflow_providers_snowflake-2.7.0-py3-none-any.whl
apache-airflow-providers-sqlite @ file:///pip-packages/apache_airflow_providers_sqlite-2.1.3-py3-none-any.whl
apache-airflow-providers-ssh==2.3.0

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
What errors do you see? you have not written that. Please clarify. I expect this is a problem with some of the templated fields of the opearator that get  a different value than string - but it's hard to say without you showing the error. My guess is that you are passing some data that looks like a Dict to a templated field and it gets rendered as a dict..

There are various ways you can deal with it - for eample you can extend SimpleHttpOperator and remove the field that causes the problem from templated_fields, pass your data in a different format, or disable retndering as native_obj,

By definition, render_as_nativeobj will render all templated_fields as native objects and the fact that it might not work in all cases is the whole reason why it is disabled by default - then you can decide in your DAG how  to deal with it on your own after you enable it. Moving it to discussions awaiting more comments, and clairfications, but it seems like ti's working as expected for now.

**Reply by subhosuper** [reply]
@potiuk  This is really important. I am facing the exact same issue except my data argument looks like this `data = {"key1": "{{ ti.xcom_pull(task_ids='first_task', key='return_value') }}" }`
I have even tried doing this `data = json.dumps({"key1": "{{ ti.xcom_pull(task_ids='first_task', key='return_value') }}" })` but doesn't help.

**Comment by subhosuper** [comment]
I have an issue open too related to the same thing https://github.com/apache/airflow/issues/26149

**Reply by potiuk** [reply]
@sd274  @subhosuper -> I proposed a workaround in #26149  that you could both test and see if it works.

**Reply by potiuk** [reply]
It's now converted to  dicsusion in https://github.com/apache/airflow/discussions/26336

**Reply by potiuk** [reply]
But if either of you have any suggestions how to solve it differently - I am all ears too.
